browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc

```diff
@@ -299,13 +299,15 @@ void BraveNewTabPageHandler::OnRemoveCustomImageBackground(
 
   auto background_pref = NTPBackgroundPrefs(profile_->GetPrefs());
   background_pref.RemoveCustomImageFromList(file_name);
-  if (background_pref.GetType() == NTPBackgroundPrefs::Type::kCustomImage &&
-      absl::get<std::string>(background_pref.GetSelectedValue()) == file_name) {
+  if (background_pref.GetType() == NTPBackgroundPrefs::Type::kCustomImage) {
     if (auto custom_images = background_pref.GetCustomImageList();
-        !custom_images.empty()) {
+        !custom_images.empty() &&
+        absl::get<std::string>(background_pref.GetSelectedValue()) ==
+            file_name) {
+      // Reset to the next candidate after we've removed the chosen one.
       background_pref.SetSelectedValue(custom_images.front());
-    } else {
-      // Reset to default
+    } else if (custom_images.empty()) {
+      // Reset to default when there's no available custom images.
```

Comment:
How about moving whole these logic (from302 to 314) to `NTPBackgroundPrefs`?

Reply:
Of course, I'm fine with this PR if we want `NTPBackgroundPrefs` as simple as possible.

Reply:
> Of course, I'm fine with this PR if we want NTPBackgroundPrefs as simple as possible.

Yes, this is what I'm more leaning toward.